airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/dest_state_lifecycle_manager/DestSingleStateLifecycleManager.java

```diff
@@ -58,6 +59,22 @@ public void markFlushedAsCommitted() {
     }
   }
 
+  @Override
+  public void markCommittedAsEmitted() {
+    if (lastCommittedState != null) {
+      lastEmittedState = lastCommittedState;
```

Comment:
where do we actually use `lastEmittedState`? afaict this field is write-only, i.e. we're holding onto this variable but never actually doing anything with it

Reply:
`lastEmittedState` is not used, it was more of design decision initially to create a category that the STATE message(s) have been committed and returned back to the platform. However, while speaking with Joe, it makes more practical sense to simply mark this value as `null` and add a comment how come it marks it as `null`
```java
// This is synonymous with saying the STATE message has been returned to the platform so we clear out the committed state
```